### PR TITLE
add products page

### PR DESCRIPTION
**Pull request by iRohitGaur**
product

Files to check: 
- [Check these files](https://github.com/iRohitGaur/StarCart/pull/9/files?file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&show-viewed-files=true)
- Products.jsx
- productsReducer.js

Files to ignore:
- all css files
- index.js files

### CODE REVIEW

File: src/pages/products/Products.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      <ProductSidebar state={state} dispatch={dispatch} />
+      <div className="product_area flex_row flex_gap2 flex_wrap">
+        {filteredData.map((product) => (
+          <Card key={product.id} product={product} btnTitle="Add to cart" />
+        ))}
+      </div>
+    </main>
+  );
+}
+
```
Comment (manojsarna) [inline src/pages/products/Products.jsx:20]: This looks good !!

File: src/reducers/productsReducer.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        return {
+          ...state,
+          [action.filterType]: state[action.filterType].includes(action.filter)
+            ? state[action.filterType].filter((type) => type !== action.filter)
+            : [...state[action.filterType], action.filter],
+        };
+      case "RESET_FILTERS":
+        return initialState;
+      default:
+        return state;
```
Comment (manojsarna) [inline src/reducers/productsReducer.js]: Suggestion : you can do error handling in the default case

File: src/reducers/productsReducer.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  if (state.bird.length !== 0) {
+    filteredData = filteredData.filter((p) => state.bird.includes(p.bird));
+  }
+
+  // RG: Filter based on Price
+  if (state.price !== initialState.price) {
+    filteredData = filteredData.filter((p) => p.price.current <= state.price);
+  }
+
+  // RG: Filter based on Rating
```
Comment (manojsarna) [inline src/reducers/productsReducer.js:51]: Rest of the code looks good!!

**Review by manojsarna — APPROVED** [review]
Overall the code looks superb!!

Do check comments for suggestions :)

File: src/reducers/productsReducer.js (hunk 2 above)
Comment (iRohitGaur) [reply src/reducers/productsReducer.js]: Done! Thanks for pointing it out.

**Review by srishtigautam21 — APPROVED** [review]
The code is well readable and understandable
The UI looks great
As always good work rohit

**Pull request merged**